src/sentry/tagstore/snuba/backend.py

```diff
@@ -16,69 +17,363 @@
 
     def get_group_tag_value(self, project_id, group_id, environment_id, key, value):
-        pass
+        from sentry.tagstore.exceptions import GroupTagValueNotFound
+        start, end = self.get_time_range()
+        tag = 'tags[{}]'.format(key)
+        filters = {
+            'project_id': [project_id],
+            'environment': [environment_id],
+        }
+        if group_id is not None:
+            filters['issue'] = [group_id]
+        conditions = [
+            [tag, '=', value]
+        ]
+        aggregations = [['count', '', 'count']]
+
+        result = snuba.query(start, end, [], conditions, filters, aggregations)
+
+        if result == 0:
+            raise GroupTagValueNotFound
+        else:
+            return ObjectWrapper({
+                'times_seen': result,
+                'key': key,
+                'value': value,
+                'group_id': group_id,
+            })
 
     def get_group_tag_values(self, project_id, group_id, environment_id, key):
-        pass
+        start, end = self.get_time_range(7)
+        tag = 'tags[{}]'.format(key)
+        filters = {
+            'project_id': [project_id],
+            'environment': [environment_id],
+        }
+        if group_id is not None:
+            filters['issue'] = [group_id]
+        conditions = [[tag, '!=', '']]
+        aggregations = [
+            ['count', '', 'count'],
+            ['min', SEEN_COLUMN, 'first_seen'],
+            ['max', SEEN_COLUMN, 'last_seen'],
+        ]
+
+        result = snuba.query(start, end, [tag], conditions, filters,
+                             aggregations)
+
+        return [ObjectWrapper({
+            'times_seen': val['count'],
+            'first_seen': val['first_seen'],
+            'last_seen': val['last_seen'],
+            'key': key,
+            'value': name,
+            'group_id': group_id,
+        }) for name, val in six.iteritems(result)]
+
+    def get_group_list_tag_value(self, project_id, group_ids, environment_id, key, value):
+        start, end = self.get_time_range()
+        tag = 'tags[{}]'.format(key)
+        filters = {
+            'project_id': [project_id],
+            'environment': [environment_id],
+            'issue': group_ids,
+        }
+        conditions = [
+            [tag, '=', value]
+        ]
+        aggregations = [['count', '', 'count']]
 
-    def get_group_list_tag_value(self, project_id, group_id_list, environment_id, key, value):
-        pass
+        result = snuba.query(start, end, ['issue'], conditions, filters, aggregations)
+
+        return {
+            issue: ObjectWrapper({
+                'times_seen': count,
+                'key': key,
+                'value': value,
+                'group_id': issue,
+            }) for issue, count in six.iteritems(result)}
 
     def get_group_tag_value_count(self, project_id, group_id, environment_id, key):
-        pass
+        start, end = self.get_time_range(7)
```

Comment:
nit: We only limit to 7 days in the non-Postgres case currently. On Postgres we limited by rows (for perf/timeouts) though, so I'm not sure what to do here off the top of my head.

Reply:
Ah yeah, may as well just conform to the postgres version so that the results are somewhat comparable when testing.